### PR TITLE
Remove Delete Method from BlobStore

### PR DESCRIPTION
**Pull request by original-brownbear**
* The delete method on the blob store was used almost nowhere and just duplicates the delete method on the blob containers
  * The fact that it provided for some recursive delete logic (that did not behave the same way on all implementations) was not used and not properly tested either
  * Had to adjust the verification cleanup method a little since that was the only user of this logic. Now it uses list and then delete to clean up the blobs it creates.

### CODE REVIEW

**Comment by original-brownbear** [comment]
From the "related note" in https://github.com/elastic/elasticsearch/pull/41368#pullrequestreview-231140163 :)

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -631,7 +631,10 @@ public String startVerification() {
     public void endVerification(String seed) {
         if (isReadOnly() == false) {
             try {
-                blobStore().delete(basePath().add(testBlobPrefix(seed)));
+                final String testPrefix = testBlobPrefix(seed);
+                final BlobContainer container = blobStore().blobContainer(basePath().add(testPrefix));
+                    container.deleteBlobsIgnoringIfNotExists(List.copyOf(container.listBlobs().keySet()));
```
Comment (andrershov) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: nit: formatting

**Review by andrershov — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/docbldesx

**Comment by original-brownbear** [comment]
thanks @andrershov !

**Pull request merged**